presto-spi/src/main/java/com/facebook/presto/spi/connector/ConnectorContext.java

```diff
@@ -0,0 +1,18 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.connector;
+
+public interface ConnectorContext
```

Comment:
I think it depends on how often we expect to have changes or how many things are needed. For example, `getTypes()` or `getBlockEncodingFactories()` seem unlikely to need new things. `ConnectorFactory` has many things including "temporary" interfaces like `PageSorter` that we want to add/remove at will.

For `Connector`, an upcoming change is a replacement for `NodeManager` (that only selects nodes for that connector). Are there many other connector-instance specific things that will be added?